app/share.tsx

```diff
@@ -117,10 +123,12 @@ class Root extends React.Component<{}, IState> {
 
 		if (currentServer) {
 			await localAuthenticate(currentServer);
-			this.setState({ root: 'inside' });
+			// @ts-ignore
+			this.mounted ? this.setState({ root: 'inside' }) : (this.state.root = 'inside');
 			await shareExtensionInit(currentServer);
 		} else {
-			this.setState({ root: 'outside' });
+			// @ts-ignore
+			this.mounted ? this.setState({ root: 'outside' }) : (this.state.root = 'outside');
```

Comment:
Can we not use inline if?
It's not as clear as a normal if and you're actually hiding what's getting ignored on that ts-ignore.